File: core/osrd-railjson/src/main/java/fr/sncf/osrd/railjson/schema/infra/trackobjects/RJSTrackObject.java
package fr.sncf.osrd.railjson.schema.infra.trackobjects;

/** An object on a RJSTrackSection. It's meant to be referenced from the section itself. */
public abstract class RJSTrackObject {
    /** Position from the beginning of the RJSTrackSection */
    public double position;

    public String track;
}

File: core/osrd-railjson/src/main/java/fr/sncf/osrd/railjson/schema/infra/trackobjects/RJSTrackObject.kt
package fr.sncf.osrd.railjson.schema.infra.trackobjects

/** An object on an RJSTrackSection. It's meant to be referenced from the section itself. */
interface RJSTrackObject {
    /** Position from the beginning of the RJSTrackSection */
    val position: Double

    /** RJSTrackSection identifier */
    val track: String
}
